internal/wrapper.go

```diff
@@ -65,25 +73,33 @@ func RunKubeStateMetricsWrapper(opts *options.Options) {
 		})
 		cfgViper.WatchConfig()
 
-		// Merge configFile values with opts so we get the CustomResourceConfigFile from config as well
-		configFile, err := os.ReadFile(filepath.Clean(file))
-		if err != nil {
-			klog.ErrorS(err, "failed to read options configuration file", "file", file)
-		}
+		if cfgViperReadInConfigErr == nil {
+			// Merge configFile values with opts so we get the CustomResourceConfigFile from config as well
+			configFile, err := os.ReadFile(filepath.Clean(file))
+			if err != nil {
+				klog.ErrorS(err, "failed to read options configuration file", "file", file)
+			}
 
-		yaml.Unmarshal(configFile, opts)
+			yaml.Unmarshal(configFile, opts)
```

Comment:
Somehow this has broken in this patch. See https://github.com/kubernetes/kube-state-metrics/issues/2803#issuecomment-3464511667 . In 2.17.0 a config file containing

```
custom_resource_config_file: custom-resource-config.yaml
```

results in an `opts.CustomResourceConfigFile` that's the empty string.

[In 2.16.0 it instead gets the correct value.](https://github.com/kubernetes/kube-state-metrics/issues/2803#issuecomment-3464541147)